### PR TITLE
Night Sky feedback and triton texture

### LINKED ISSUE #1813: Cardinal Directions for Globes
Users would desire to show the cardinal directions, NESW, on globes such as exampled in the attached video. 

This is a feature used by a lot of planetariums when showing the night sky.

The example was made with a RenderableSphere for a static location.

Different approaches could be taken to implement this:
- A feature of the RenderableGlobe which knows the camera position in relation to globe and can draw them with the FontRenderer. Potential issue with this could be that maybe you would be required to be focused on the globe.
- This could be done with a RenderableSphere or even some RenderableModel's. Issues with this would be that it would only work when at a fixed location on the ground (as demonstrated in png, however maybe this is a 'feet on the ground' feature and so it could fade out when elevation is changed. This would require a way of updating the position dynamically.   

https://user-images.githubusercontent.com/708535/149025952-25535931-658e-46f9-9f5d-95d1ae66c172.mp4

OpenSpace_000000

### LINKED ISSUE #2198: Add Triton texture
From Carter: https://astrogeology.usgs.gov/search/map/Triton/Mosaic/Triton_Voyager2_ClrMosaic_GlobalFill_600m

for use with voyager profile

### LINKED ISSUE #2648: NightSky feedback
Feedback items from NS reivew:

- [x] toggle buttons for hide/show
- [x] horizonal glow
- [x] NS moon
- [x] venus BB brighter
- [x] Ecliptic band size could match orbit maxima
- [x] galactic band should be named galatcic equator line
- [ ] ecliptic band could get tick marks added
- [x] extend size of lightpollution out past earth
- [x] NESW + NESW band should use same font

### PR DESCRIPTION
**Pull request by micahnyc**
closes #2198
closes #1813 
closes #2648

### CODE REVIEW

**Comment by micahnyc** [comment]
[cardinal_directions_textures.zip](https://github.com/OpenSpace/OpenSpace/files/11647956/cardinal_directions_textures.zip)

[ecliptic_band_textures.zip](https://github.com/OpenSpace/OpenSpace/files/11647958/ecliptic_band_textures.zip)

File: data/assets/nightsky/altaz.asset [hunk 1]
```diff
@@ -58,35 +58,42 @@ local AltAzGrid = {
...
     openspace.setPropertyValueSingle("Scene.AltAzGridPosition.Translation.Globe", camera.Anchor)
-    openspace.setPropertyValueSingle("Scene.AltAzGridPosition.Translation.Latitude", lat)
-    openspace.setPropertyValueSingle("Scene.AltAzGridPosition.Translation.Longitude", lon)
     openspace.setPropertyValueSingle("Scene.AltAzGridPosition.Rotation.Globe", camera.Anchor)
-    openspace.setPropertyValueSingle("Scene.AltAzGridPosition.Rotation.Latitude", lat)
-    openspace.setPropertyValueSingle("Scene.AltAzGridPosition.Rotation.Longitude", lon)
-    openspace.setPropertyValueSingle("Scene.AltAzGrid.Renderable.Enabled", true)
-    openspace.setPropertyValueSingle("Scene.AltAzGrid.Renderable.Fade", 0.0)
-    openspace.setPropertyValueSingle("Scene.AltAzGrid.Renderable.Fade", 1.0, 1.0)
+    openspace.fadeIn("Scene.AltAzGrid.Renderable");
```
Comment (alexanderbock) [inline data/assets/nightsky/altaz.asset:70]:
```suggestion
    openspace.fadeIn("Scene.AltAzGrid.Renderable")
```

File: data/assets/nightsky/altaz.asset [hunk 2]
```diff
@@ -58,35 +58,42 @@ local AltAzGrid = {
...
   Documentation = "Shows a local Altitude/Azimuth grid centered around your position",
-  GuiPath = "/Night Sky/Lines and Grids",
+  GuiPath = "/Night Sky/Lines and Grids/Show and Hide",
   IsLocal = false
 }
 
 local HideAltaz = {
   Identifier = "os.nightsky.HideAltaz",
   Name = "Hide Alt/Az grid",
+  Command = 'openspace.fadeOut("Scene.AltAzGrid.Renderable")',
```
Comment (alexanderbock) [inline data/assets/nightsky/altaz.asset:80]:
```suggestion
  Command = "openspace.fadeOut('Scene.AltAzGrid.Renderable')",
```

File: data/assets/nightsky/altaz.asset [hunk 3]
```diff
@@ -58,35 +58,42 @@ local AltAzGrid = {
...
   Command = [[
-    openspace.setPropertyValueSingle("Scene.AltAzGrid.Renderable.Fade", 0, 1)
-  ]],
-  Documentation = "Hides the Alt/Az grid",
+    local camera = openspace.navigation.getNavigationState();
+    openspace.setParent('AltAzGridPosition', camera.Anchor)
+    openspace.setPropertyValueSingle('Scene.AltAzGridPosition.Translation.Globe', camera.Anchor);
+    openspace.setPropertyValueSingle('Scene.AltAzGridPosition.Rotation.Globe', camera.Anchor);
+    openspace.toggleFade("Scene.AltAzGrid.Renderable")
+    ]],
```
Comment (alexanderbock) [inline data/assets/nightsky/altaz.asset:95]:
```suggestion
    local camera = openspace.navigation.getNavigationState()
    openspace.setParent("AltAzGridPosition", camera.Anchor)
    openspace.setPropertyValueSingle("Scene.AltAzGridPosition.Translation.Globe", camera.Anchor)
    openspace.setPropertyValueSingle("Scene.AltAzGridPosition.Rotation.Globe", camera.Anchor)
    openspace.toggleFade("Scene.AltAzGrid.Renderable")
  ]],
```

File: data/assets/nightsky/cardinal_directions.asset [hunk 4]
```diff
@@ -76,44 +69,58 @@ local showCommand = [[
...
+  IsLocal = false
+}
+
 local HideNesw = {
   Identifier = "os.nightsky.HideNesw",
   Name = "Hide cardinal directions",
-  Command = [[
-    openspace.setPropertyValueSingle("Scene.CardinalDirectionSphere.Renderable.Fade", 0.0, 1.0)
-  ]],
+  Command = 'openspace.fadeOut("Scene.CardinalDirectionSphere.Renderable")',
```
Comment (alexanderbock) [inline data/assets/nightsky/cardinal_directions.asset:123]:
```suggestion
  Command = "openspace.fadeOut('Scene.CardinalDirectionSphere.Renderable')",
```

File: data/assets/nightsky/ecliptic_band.asset [hunk 5]
```diff
@@ -72,23 +72,26 @@ local EclipticBand = {
 local ShowEclipticLine = {
   Identifier = "os.nightsky.ShowEclipticLine",
   Name = "Show ecliptic line",
-  Command = [[
-    openspace.setPropertyValueSingle("Scene.EclipticLine.Renderable.Fade", 0.0)
-    openspace.setPropertyValueSingle("Scene.EclipticLine.Renderable.Enabled", true)
-    openspace.setPropertyValueSingle("Scene.EclipticLine.Renderable.Fade", 1.0, 1.0)
-  ]],
+  Command = 'openspace.fadeIn("Scene.EclipticLine.Renderable")',
```
Comment (alexanderbock) [inline data/assets/nightsky/ecliptic_band.asset:75]:
```suggestion
  Command = "openspace.fadeIn('Scene.EclipticLine.Renderable')",
```

File: data/assets/nightsky/ecliptic_band.asset [hunk 6]
```diff
@@ -72,23 +72,26 @@ local EclipticBand = {
...
   IsLocal = false
 }
 
 local HideEclipticLine = {
   Identifier = "os.nightsky.HideEclipticLine",
   Name = "Hide ecliptic line",
-  Command = [[
-    openspace.setPropertyValueSingle("Scene.EclipticLine.Renderable.Fade", 0.0, 1.0)
-  ]],
+  Command = 'openspace.fadeOut("Scene.EclipticLine.Renderable")',
```
Comment (alexanderbock) [inline data/assets/nightsky/ecliptic_band.asset:84]:
```suggestion
  Command = "openspace.fadeOut('Scene.EclipticLine.Renderable')",
```

File: data/assets/nightsky/ecliptic_band.asset [hunk 7]
```diff
@@ -97,22 +100,29 @@ local ShowEclipticBand = {
...
   ]],
   Documentation = "Hides the ecliptic band",
+  GuiPath = "/Night Sky/Lines and Grids/Show and Hide",
+  IsLocal = false
+}
+
+local ToggleEclipticBand = {
+  Identifier = "os.nightsky.ToggleEclipticBand",
+  Name = "Toggle ecliptic band",
+  Command = 'openspace.toggleFade("Scene.EclipticBand.Renderable")',
```
Comment (alexanderbock) [inline data/assets/nightsky/ecliptic_band.asset:124]:
```suggestion
  Command = "openspace.toggleFade('Scene.EclipticBand.Renderable')",
```

File: data/assets/nightsky/equatorial_band.asset [hunk 8]
```diff
@@ -41,22 +41,29 @@ local ShowEquatorialLine = {
...
   ]],
   Documentation = "Hides the equatorial line",
+  GuiPath = "/Night Sky/Lines and Grids/Show and Hide",
+  IsLocal = false
+}
+
+local ToggleEquatorialLine = {
+  Identifier = "os.nightsky.ToggleEquatorialLine",
+  Name = "Toggle equatorial line",
+  Command = 'openspace.toggleFade("Scene.EquatorialLine.Renderable")',
```
Comment (alexanderbock) [inline data/assets/nightsky/equatorial_band.asset:65]:
```suggestion
  Command = "openspace.toggleFade('Scene.EquatorialLine.Renderable')",
```

File: data/assets/nightsky/galactic_band.asset [hunk 9]
```diff
@@ -31,26 +31,34 @@ local ShowGalacticBand = {
...
+  GuiPath = "/Night Sky/Lines and Grids/Show and Hide",
   IsLocal = false
 }
 
+local ToggleGalacticBand = {
+  Identifier = "os.nightsky.ToggleGalacticBand",
+  Name = "Toggle galactic equator line",
+  Command = [[
+    openspace.toggleFade("Scene.GalacticBand.Renderable")
+  ]],
```
Comment (alexanderbock) [inline data/assets/nightsky/galactic_band.asset:57]:
```suggestion
  Command = "openspace.toggleFade('Scene.GalacticBand.Renderable')",
```

File: data/assets/nightsky/meridian.asset [hunk 10]
```diff
@@ -58,31 +58,40 @@ local ShowMeridian = {
...
+local ToggleMeridian = {
+  Identifier = "os.nightsky.ToggleMeridian",
+  Name = "Toggle local meridian",
+  Command = [[
+    local camera = openspace.navigation.getNavigationState();
+    openspace.setParent('MeridianPosition', camera.Anchor)
+    openspace.setPropertyValueSingle('Scene.MeridianPosition.Translation.Globe', camera.Anchor);
+    openspace.setPropertyValueSingle('Scene.MeridianPosition.Rotation.Globe', camera.Anchor);
+    openspace.toggleFade("Scene.MeridianPlane.Renderable")
+    ]],
```
Comment (alexanderbock) [inline data/assets/nightsky/meridian.asset:93]:
```suggestion
    local camera = openspace.navigation.getNavigationState()
    openspace.setParent("MeridianPosition", camera.Anchor)
    openspace.setPropertyValueSingle("Scene.MeridianPosition.Translation.Globe", camera.Anchor)
    openspace.setPropertyValueSingle("Scene.MeridianPosition.Rotation.Globe", camera.Anchor)
    openspace.toggleFade("Scene.MeridianPlane.Renderable")
  ]],
```

File: data/assets/nightsky/planets.asset [hunk 11]
```diff
@@ -118,23 +118,14 @@ local ShowNightSkyPlanets = {
...
-    openspace.setPropertyValueSingle("Scene.NightSkySaturn.Renderable.Fade", 1.0, 0.25)
+    openspace.fadeIn("Scene.NightSkyMercury.Renderable")
+    openspace.fadeIn("Scene.NightSkyVenus.Renderable")
+    openspace.fadeIn("Scene.NightSkyMars.Renderable")
+    openspace.fadeIn("Scene.NightSkyJupiter.Renderable")
+    openspace.fadeIn("Scene.NightSkySaturn.Renderable")
+    openspace.setPropertyValueSingle("Scene.Moon.Scale.Scale", 2)
   ]],
-  Documentation = "Show night sky versions of the planets",
+  Documentation = "Show night sky versions of the planets (note increases Scale of Moon)",
```
Comment (alexanderbock) [inline data/assets/nightsky/planets.asset:128]:
```suggestion
  Documentation = "Show night sky versions of the planets (Note: Increases the scale of the Moon)",
```

File: data/assets/nightsky/planets.asset [hunk 12]
```diff
@@ -143,18 +134,40 @@ local HideNightSkyPlanets = {
...
+    openspace.toggleFade("Scene.NightSkyJupiter.Renderable")
+    openspace.toggleFade("Scene.NightSkySaturn.Renderable")
+    local scale = openspace.getPropertyValue("Scene.Moon.Scale.Scale")
+    if (scale > 1) then
+      openspace.setPropertyValueSingle("Scene.Moon.Scale.Scale", 1.0)
+    else
+      openspace.setPropertyValueSingle("Scene.Moon.Scale.Scale", 2.0)
+    end
+  ]],
+  Documentation = "Toggles visibility of the night sky versions of the planets (note increases Scale of Moon)",
```
Comment (alexanderbock) [inline data/assets/nightsky/planets.asset:166]:
```suggestion
  Documentation = "Toggles visibility of the night sky versions of the planets (Note: Increases the scale of the Moon)",
```

File: data/assets/nightsky/zenith.asset [hunk 13]
```diff
@@ -66,31 +66,39 @@ local ShowZenith = {
...
+local ToggleZenith = {
+  Identifier = "os.nightsky.ToggleZenith",
+  Name = "Toggle local zenith",
+  Command = [[
+    local camera = openspace.navigation.getNavigationState();
+    openspace.setParent("ZenithPosition", camera.Anchor)
+    openspace.setPropertyValueSingle("Scene.ZenithPosition.Translation.Globe", camera.Anchor);
+    openspace.setPropertyValueSingle("Scene.ZenithPosition.Rotation.Globe", camera.Anchor);
+    openspace.toggleFade("Scene.ZenithDot.Renderable")
+    ]],
```
Comment (alexanderbock) [inline data/assets/nightsky/zenith.asset:100]:
```suggestion
    local camera = openspace.navigation.getNavigationState()
    openspace.setParent("ZenithPosition", camera.Anchor)
    openspace.setPropertyValueSingle("Scene.ZenithPosition.Translation.Globe", camera.Anchor)
    openspace.setPropertyValueSingle("Scene.ZenithPosition.Rotation.Globe", camera.Anchor)
    openspace.toggleFade("Scene.ZenithDot.Renderable")
  ]],
```

File: data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset [hunk 14]
```diff
@@ -0,0 +1,42 @@
+local globeIdentifier = asset.require("scene/solarsystem/planets/neptune/triton/triton").Triton.Identifier
```
Comment (alexanderbock) [inline data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset:1]:
```suggestion
local globe = asset.require("../triton")


```

File: data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+
+local texturesPath = asset.syncedResource({
+  Name = "Triton Textures",
+  Type = "HttpSynchronization",
+  Identifier = "triton_textures",
+  Version = 1
+})
+
```
Comment (alexanderbock) [inline data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset:9]:
```suggestion


```

File: data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+local layer = {
+  Identifier = "Triton_Voyager2_ClrMosaic_GlobalFill_600m",
+  Name = "Triton Voyager2 Mosaic [local]",
+  Enabled = asset.enabled,
+  FilePath = texturesPath .. "Triton_Voyager2_ClrMosaic_GlobalFill_600m.png",
+  Description = [[Voyager 2 data was used to construct the best-ever global color map of Triton.
+  This map has a resolution of 1,970 feet (600 meters per pixel [m]). Color was synthesized by
+  combining high-resolution images taken through orange, violet, and ultraviolet filters; these
+  images were displayed as red, green, and blue images and combined to create this
+  color version (Smith et al., 1989).]],
```
Comment (alexanderbock) [inline data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset:19]:
```suggestion
  Description = [[Voyager 2 data was used to construct the best-ever global color map of Triton.
    This map has a resolution of 1,970 feet (600 meters per pixel [m]). Color was synthesized by
    combining high-resolution images taken through orange, violet, and ultraviolet filters; these
    images were displayed as red, green, and blue images and combined to create this
    color version (Smith et al., 1989).]]
```

File: data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+asset.onInitialize(function()
```
Comment (alexanderbock) [inline data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset:22]:
```suggestion

asset.onInitialize(function()
```

File: data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+  openspace.globebrowsing.addLayer(globeIdentifier, "ColorLayers", layer)
```
Comment (alexanderbock) [inline data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset:23]:
```suggestion
  openspace.globebrowsing.addLayer(globe.Triton.Identifier, "ColorLayers", Layer)
```

File: data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+end)
+
+asset.onDeinitialize(function()
+  openspace.globebrowsing.deleteLayer(globeIdentifier, "ColorLayers", layer)
```
Comment (alexanderbock) [inline data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset:27]:
```suggestion
  openspace.globebrowsing.deleteLayer(globe.Triton.Identifier, "ColorLayers", Layer)
```

File: data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset [hunk 20]
```diff
@@ -0,0 +1,42 @@
+local globeIdentifier = asset.require("scene/solarsystem/planets/neptune/triton/triton").Triton.Identifier
+
+local texturesPath = asset.syncedResource({
+  Name = "Triton Textures",
+  Type = "HttpSynchronization",
+  Identifier = "triton_textures",
+  Version = 1
+})
+
+local layer = {
```
Comment (alexanderbock) [inline data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset:10]:
```suggestion
local Layer = {
```

File: data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+
+asset.onInitialize(function()
+  openspace.globebrowsing.addLayer(globeIdentifier, "ColorLayers", layer)
+end)
+
+asset.onDeinitialize(function()
+  openspace.globebrowsing.deleteLayer(globeIdentifier, "ColorLayers", layer)
+end)
+
+asset.export("layer", layer)
```
Comment (alexanderbock) [inline data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset:30]:
```suggestion
asset.export("Layer", Layer)
```

File: data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset [hunk 22]
```diff
@@ -0,0 +1,42 @@
...
+
+
+
+asset.meta = {
+  Name = "Triton Voyager2 Mosaic",
+  Version = "1.0",
+  Description = [[Voyager2 mosaic of Triton. This layer local in the sync folder.]],
```
Comment (alexanderbock) [inline data/assets/scene/solarsystem/planets/neptune/triton/layers/colorlayers/Triton_Voyager2_ClrMosaic_GlobalFill_600m.asset:37]:
```suggestion
  Description = [[Voyager2 mosaic of Triton. This layer is local in the sync folder.]],
```

File: data/assets/scene/solarsystem/planets/neptune/triton/trail.asset [hunk 23]
```diff
@@ -0,0 +1,46 @@
+asset.require("spice/base")
+local transforms = asset.require("../transforms")
+local kernel = asset.require("../kernels").nep097
```
Comment (alexanderbock) [inline data/assets/scene/solarsystem/planets/neptune/triton/trail.asset:3]:
```suggestion
local kernel = asset.require("../kernels")


```

File: data/assets/scene/solarsystem/planets/neptune/triton/trail.asset [hunk 24]
```diff
@@ -0,0 +1,46 @@
...
+local Trail = {
+  Identifier = "TritonTrail",
+  Parent = transforms.NeptuneBarycenter.Identifier,
+  Renderable = {
+    Type = "RenderableTrailOrbit",
+    Translation = {
+      Type = "SpiceTranslation",
+      Target = "TRITON",
+      Observer = "NEPTUNE BARYCENTER",
+      Kernels = kernel
```
Comment (alexanderbock) [inline data/assets/scene/solarsystem/planets/neptune/triton/trail.asset:14]:
```suggestion
      Observer = "NEPTUNE BARYCENTER"
```

File: data/assets/scene/solarsystem/planets/neptune/triton/trail.asset [hunk 25]
```diff
@@ -0,0 +1,46 @@
...
+  },
+  Tag = { "moonTrail_solarSystem", "moonTrail_giants", "moonTrail_neptune" },
+  GUI = {
+    Name = "Triton Trail",
+    Path = "/Solar System/Planets/Neptune/Major Moons/Triton"
+  }
+}
+
+asset.onInitialize(function()
+  openspace.addSceneGraphNode(Trail)
```
Comment (alexanderbock) [inline data/assets/scene/solarsystem/planets/neptune/triton/trail.asset:28]:
```suggestion
  openspace.spice.loadKernel(kernel.nep097)


  openspace.addSceneGraphNode(Trail)
```

File: data/assets/scene/solarsystem/planets/neptune/triton/trail.asset [hunk 26]
```diff
@@ -0,0 +1,46 @@
...
+end)
+
+asset.onDeinitialize(function()
+  openspace.removeSceneGraphNode(Trail)
+end)
```
Comment (alexanderbock) [inline data/assets/scene/solarsystem/planets/neptune/triton/trail.asset:33]:
```suggestion


  openspace.spice.unloadKernel(kernel.nep097)
end)
```

File: data/assets/scene/solarsystem/planets/neptune/triton/triton.asset [hunk 27]
```diff
@@ -100,8 +105,7 @@ asset.export(TritonLabel)
 asset.meta = {
   Name = "Triton",
   Version = "1.1",
-  Description = [[Procedural Globe asset containing Neptune's moon: Triton. A blank globe
-    and SPICE trail are generated for the moon]],
+  Description = [[Globe asset containing Neptune's moon: Triton]],
```
Comment (alexanderbock) [inline data/assets/scene/solarsystem/planets/neptune/triton/triton.asset:108]:
```suggestion
  Description = [[Globe asset containing Neptune's moon Triton]],
```

File: scripts/core_scripts.lua [hunk 28]
```diff
@@ -87,6 +87,20 @@ openspace.documentation = {
...
+            identifier = "String",
+            fadeTime = "Number?",
+            endScript = "String?"
+        },
+        Documentation = "Toggles the fade state of the node(s) with the given identifier over the given " ..
+        "time in seconds. The identifier can contain a tag and/or a wildcard to target " ..
+        "several nodes. If the fade time is not provided then the " ..
+        "'OpenSpaceEngine.FadeDuration' property will be used instead. If the third " ..
+        "argument (endScript) is provided then that script will be run after the fade " ..
+        "is finished."
```
Comment (alexanderbock) [inline scripts/core_scripts.lua:103]:
```suggestion
        Documentation = [[Toggles the fade state of the node(s) with the given identifier over the given
          time in seconds. The identifier can contain a tag and/or a wildcard to target
          several nodes. If the fade time is not provided then the
          "OpenSpaceEngine.FadeDuration" property will be used instead. If the third
          argument (endScript) is provided then that script will be run after the fade
          is finished.]]
```

File: scripts/core_scripts.lua [hunk 29]
```diff
@@ -302,3 +316,20 @@ openspace.fadeOut = function(identifier, fadeTime, endScript)
         )
     end
 end
+
+openspace.toggleFade = function(renderable, fadeTime, endScript)
+  if (fadeTime == nil) then
+    fadeTime = openspace.getPropertyValue("OpenSpaceEngine.FadeDuration")
+  end
+  local enabled = openspace.getPropertyValue(renderable .. ".Enabled");
+  local fadeState = openspace.getPropertyValue(renderable .. ".Fade");
```
Comment (alexanderbock) [inline scripts/core_scripts.lua:325]:
```suggestion
  local enabled = openspace.getPropertyValue(renderable .. ".Enabled")
  local fadeState = openspace.getPropertyValue(renderable .. ".Fade")
```

File: scripts/core_scripts.lua [hunk 30]
```diff
@@ -302,3 +316,20 @@ openspace.fadeOut = function(identifier, fadeTime, endScript)
...
+  if (enabled == true) then
```
Comment (alexanderbock) [inline scripts/core_scripts.lua:326]:
```suggestion
  if (enabled) then
```

**Comment by alexanderbock** [comment]
Files are on the server

**Comment by alexanderbock** [comment]
Another comment, this PR also adds a new Lua function `openspace.toggleFade`.  The squashed commit message should reflect this.

**Comment by alexanderbock** [comment]
Closes #2258

**Pull request merged**